### ISSUE TITLE
Integrate data-sync layer

### ISSUE THREAD

**Issue by cammellos** [status-im/status-protocol-go#7]
As a user
I want my messages to be more reliable 
So that I have more confidence in the app

### Description

This tasks involves integrating https://github.com/status-im/mvds in the actual codebase, according to https://github.com/status-im/bigbrother-specs/blob/master/data_sync/mvds.md#secure-transport.

It should work for one-to-one and private group chats. Pubic chats are out of scope, although a non-breaking path forward should be indicated.

Messages should be authenticated to the original sender.
Messages should be authorized so that only peers that should have access to the messages are able to see/request messages.
Any messages exchanged by the data-sync layer should be PFS encrypted for private communication.

Breaking changes are ok, as long as it's not blocking any possible release pre-v1.

### Acceptance criteria

- MVDS is integrated into the app
- 1-to-1 and group chat messages are exchanged using MVDS
- Messages are authenticated/authorized correctly when relaying them to third party or sending them directly
- A clear non-breaking path forward is indicated for public chats
- This likely will include a change in the actual protocol exchanged on the wire, specs should be updated accordingly before and after implementation: https://github.com/status-im/bigbrother-specs

#### out of scope
- Public chats
- Resource consumption tests

Depends on #2

**Comment by StatusWrike** [comment]
➤ Oskar Thoren commented:

test comment see if wrike syncs

**Comment by StatusWrike** [comment]
➤ Oskar Thoren commented:

Should be synced with https://github.com/status-im/status-protocol-go/issues/7

**Comment by StatusWrike** [comment]
➤ Wrike Bot commented:

Andrea Piana All the necessary task dependencies have been completed, so you can start working on it right away.

**Comment by cammellos** [comment]
> A clear non-breaking path forward is indicated for public chats

Currently datasync layer is between the application and encryption layer, so sending a message over a public channel with datasync is non breaking.
